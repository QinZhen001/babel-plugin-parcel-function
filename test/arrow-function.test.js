const {testPlugin, removeSpace} = require("./util/util")

const code1 = `async () => {
    await aaa();
  };
`

const code2 = `
const success = (res) => {
    console.log(res, '跳转成功');
    return true
  }
`


// -------------------------------------------

describe("箭头函数", () => {
  test("匿名", () => {
    const res = testPlugin(code1, [], [], {})
    expect(res).toMatchSnapshot();
  })
  test("赋值", () => {
    const res = testPlugin(code2, [], [], {})
    expect(res).toMatchSnapshot();
  })
})


