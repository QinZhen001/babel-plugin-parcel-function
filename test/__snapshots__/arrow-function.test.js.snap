// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`箭头函数 匿名 1`] = `
"var _this = this;

async () => {
    if (global.hotUpdate) {
        global.hotUpdate.call(_this);
    } else {
        await aaa();
    }
};"
`;

exports[`箭头函数 赋值 1`] = `
"var _this = this;

const success = res => {
    if (global.hotUpdate) {
        global.hotUpdate.call(_this, res);
    } else {
        console.log(res, '跳转成功');
        return true;
    }
};"
`;
